pkg/apis/ceph.rook.io/v1/object.go

```diff
@@ -88,6 +91,33 @@ func ValidateObjectSpec(gs *CephObjectStore) error {
 	if gs.Spec.Gateway.Port <= 0 && gs.Spec.Gateway.SecurePort <= 0 {
 		return errors.New("invalid create: either of port or securePort fields should be not be zero")
 	}
+
+	// check hosting spec
+	if gs.Spec.Hosting != nil {
+		if gs.Spec.Hosting.AdvertiseEndpoint != nil {
+			ep := gs.Spec.Hosting.AdvertiseEndpoint
+			errList := validation.IsDNS1123Subdomain(ep.DnsName)
+			if len(errList) > 0 {
+				return errors.Errorf("hosting.advertiseEndpoint.dnsName %q must be a valid DNS-1123 subdomain: %v", ep.DnsName, errList)
+			}
+			if ep.Port < 1 || ep.Port > 65535 {
```

Comment:
CRD validation gets removed or bugged. It's probably not necessary, but the other ports in the same CRs are checked, so it seemed good to follow suit here just in case.